src/server/player_sao.cpp

```diff
@@ -545,10 +564,21 @@ void PlayerSAO::disconnected()
 void PlayerSAO::unlinkPlayerSessionAndSave()
```

Comment:
For some reason unlinkPlayerSessionAndSave isn't called on player leaving. PlayerSAO::disconnected probably?

Reply:
I added some simple `std::cout` lines to `disconnected()` and `unlinkPlayerSessionAndSave()` each.
Server started from main menu: both functions are called for all players
Localplayer: only `unlinkPlayerSessionAndSave()` is called

If anything, `unlinkPlayerSessionAndSave()` is the function that is called reliably.

Reply:
For me if I connect remotely unlinkPlayerSessionAndSave is called on the client side. It's not what we need

Reply:
Would you please be so nice to explain what you mean with "client side"? PlayerSAO is a ServerActiveObject, hence has nothing to do with the client.

Reply:
Sorry, my mistake. Looks fine to me then